tests/unit/test_variants.py

```diff
@@ -70,3 +73,112 @@ def testPhasesetZero(self):
+    """
+    def setUp(self):
+        self._vsname = "testVariantSet"
+        self._backend = backend.AbstractBackend()
+        self._dataset = datasets.AbstractDataset(self._backend)
+        self._variantSet = variants.AbstractVariantSet(
+            self._dataset, self._vsname)
+
+        # create some simulated variants with dummy data
+        self._simulatedVariantSet = variants.SimulatedVariantSet(
+            self._dataset, self._vsname, randomSeed=0,
+            numCalls=1, variantDensity=1)
+        self._simulatedVariants = self._simulatedVariantSet.getVariants(
+            'ref', 100, 110, ["nope"])
+        self._realVariant = list(self._simulatedVariants)[0]
+
+    def testAddOneCallSet(self):
+        self.assertEqual(self._variantSet.getNumCallSets(), 0)
+        callSetName = "callSetName"
+        self._variantSet.addCallSet(callSetName)
+        self.assertEqual(self._variantSet.getNumCallSets(), 1)
+        callSet = self._variantSet.getCallSetByIndex(0)
+        self.assertEqual(
+            self._variantSet.getCallSetByName(callSetName), callSet)
+        self.assertEqual(
+            self._variantSet.getCallSet(callSet.getId()), callSet)
+        self.assertEqual(self._variantSet.getCallSets(), [callSet])
+
+    def testAddMultipleCallSets(self):
+        self.assertEqual(self._variantSet.getNumCallSets(), 0)
+        callSetName = "callSetName{}"
+        callSetCount = 10
+        allCallSets = []
+        for i in range(callSetCount):
+            self._variantSet.addCallSet(callSetName.format(i))
+            callSet = self._variantSet.getCallSetByIndex(i)
+            self.assertEqual(
+                self._variantSet.getCallSetByName(
+                    callSetName.format(i)), callSet)
+            self.assertEqual(
+                self._variantSet.getCallSet(callSet.getId()), callSet)
+            allCallSets.append(callSet)
+
+        self.assertEqual(self._variantSet.getNumCallSets(), callSetCount)
+        self.assertEqual(self._variantSet.getCallSets(), allCallSets)
+
+    def testCallSetMethods(self):
+        # test invalid lookups
+        self.assertRaises(IndexError, self._variantSet.getCallSetByIndex, 10)
+
+        self.assertRaises(exceptions.CallSetNameNotFoundException,
+                          self._variantSet.getCallSetByName, 'noname')
+
+        self.assertRaises(exceptions.CallSetNameNotFoundException,
+                          self._variantSet.getCallSetByName, None)
+
+        self.assertRaises(exceptions.CallSetNotFoundException,
+                          self._variantSet.getCallSet, 617)
+
+        self.assertRaises(exceptions.CallSetNotFoundException,
+                          self._variantSet.getCallSet, None)
+
+        self.assertRaises(TypeError, self._variantSet.addCallSet,
+                          ['a list of', 2])
+
+        self.assertRaises(NotImplementedError,
+                          self._variantSet.getNumVariants)
+
+    def testGetVariantId(self):
```

Comment:
I think the next two tests are aimed at the HtslibVariantSet class, and are tested elsewhere. Here, we really just want to focus on AbstractVariantSet, so we can remove them.